packages/zendesk-adapter/src/change_validators/brand_creation.ts

```diff
@@ -13,22 +13,73 @@
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
-import { ChangeValidator, getChangeData,
-  isAdditionChange, isInstanceElement } from '@salto-io/adapter-api'
+import {
+  ChangeError,
+  ChangeValidator, getChangeData, InstanceElement,
+  isAdditionChange, isInstanceElement,
+} from '@salto-io/adapter-api'
+import _ from 'lodash'
+import { collections } from '@salto-io/lowerdash'
+import { logger } from '@salto-io/logging'
 import { BRAND_TYPE_NAME } from '../constants'
+import ZendeskClient from '../client/client'
 
-export const brandCreationValidator: ChangeValidator = async changes => (
-  changes
-    .filter(isAdditionChange)
-    .map(getChangeData)
-    .filter(isInstanceElement)
-    .filter(instance => instance.elemID.typeName === BRAND_TYPE_NAME)
-    .flatMap(instance => (
-      [{
-        elemID: instance.elemID,
-        severity: 'Warning',
-        message: 'Verify brand subdomain uniqueness',
-        detailedMessage: `Brand subdomains are globally unique, please make sure to set an available subdomain for brand ${instance.value.name} before attempting to create it from Salto`,
-      }]
-    ))
-)
+const { awu } = collections.asynciterable
+const log = logger(module)
+
+type ValidRes = {
+  success: boolean
+}
+
+const isValidRes = (res: unknown): res is ValidRes => _.isObject(res) && ('success' in res)
+
+
+const invalidSubdomain = async (brand: InstanceElement, client: ZendeskClient): Promise<boolean | undefined> => {
+  if (brand.value.subdomain === undefined) {
+    return true
+  }
+  let res
+  try {
+    res = (await client.getSinglePage({
+      url: `/api/v2/accounts/available.json?subdomain=${brand.value.subdomain}`,
+    })).data
+  } catch (e) {
+    log.error(e)
+    res = undefined
+  }
+  if (res === undefined || !isValidRes(res)) {
+    return undefined
+  }
+  return !res.success
+}
+
+/**
+ * this validator checks if the subdomain of the brand is globally uniq
+ */
+export const brandCreationValidator: (client: ZendeskClient) =>
+  ChangeValidator = client => async changes => {
+    const brandAddition = await awu(changes)
+      .filter(isAdditionChange)
+      .map(getChangeData)
+      .filter(isInstanceElement)
+      .filter(instance => instance.elemID.typeName === BRAND_TYPE_NAME)
+      .map(async instance => ({ instance, invalid: await invalidSubdomain(instance, client) }))
+      .toArray()
+
+    return brandAddition
+      .filter(brandDetails => brandDetails.invalid || brandDetails.invalid === undefined)
+      .map(({ instance, invalid }): ChangeError => {
+        if (invalid === undefined) {
+          return { elemID: instance.elemID,
+            severity: 'Warning',
+            message: 'Verify brand subdomain uniqueness',
+            detailedMessage: `Brand subdomains are globally unique, please make sure to set an available subdomain for brand ${instance.value.name} before attempting to create it from Salto` }
```

Comment:
I think we can make this more actionable - e.g. something like
```suggestion
            detailedMessage: `Brand subdomains are globally unique, and we were unable to check the uniqueness of the subdomain '${...}' used for brand ${instance.value.name}. Please ensure its uniqueness before deploying` }
```